app/services/create_group.rb

```diff
@@ -0,0 +1,9 @@
+class CreateGroup
+  def self.call(create_params, leader)
+    group = Group.create(create_params)
+    membership = CreateMembership.call(group, leader)
+    membership.posts.where(post_type_id: PostType::DEFAULT_POST_ID).destroy_all
+    CreatePost.call(group, membership, PostType::LEADER_POST_ID)
```

Comment:
First of all please rebase the branch to the master, this `CreateMembership` does not exist anymore. Also instead of using `CreateMembership` (or the new `Membership::CreateService`) and deleting the default post, it seems to be much more straightforward to create the membership manually here and to add the leader post. In this case I think it's better to be less DRY, make the understanding easier, later on it can be improved and refactored when we find a fitting and reasonable way to do it.

Also other important thing, please wrap the whole method into transaction (`ActiveRecord::Base.transaction`) to avoid possible inconsistency.

Reply:
I rebased this branch to the master. I'm not sure If I understood you correctly, but i moved the leader post creation logic to the controller. I think this makes the CreateGroup service unnecessary,  since it is same as `Group.create`.